docs/guides/editor_features/watching.md

```diff
@@ -11,92 +11,142 @@ have the changes automatically reflected in your browser.
+
+Top-level functions are added to a notebook using the `@app.function`
+decorator, and classes with `@app.class_definition`; these appear in your
+notebook as cells with just a function or class definition. These functions and
+classes must be pure, closing over only other pure functions and classes, or
+imports and constants defined in an `app.setup` `with` block.
+
+Here is a complete example that you can copy/paste and run locally:
 
-For more details see the [library guide](../reusing_functions.md).
 
 ```python
+import marimo
+
+app = marimo.App()
+
 with app.setup:
+    # These symbols can be used by top-level functions and classes
+    # (as well as by regular cells)
+    import numpy as np
+
     CONSTANT: int = 1
-    import marimo as mo
 
 @app.function
-def my_function(x): ...
+def my_function(x: np.ndarray):
+    return np.mean(x) + CONSTANT
 
 @app.class_definition
-class MyClass: ...
+class MyClass:
+    ...
+
+@app.cell
+def _():
+    my_function(np.random.randn(2, 2))
+    return
+
+if __name__ == "__main__":
+    app.run()
 ```
 
-!!! question "Want to use your own LSP for typing?"
+For more details see the [guide on reusable functions and classes](../reusing_functions.md).
+
+### Types and autocompletion
 
-Explicitly typing your definitions will let marimo carry the annotations into
-function signatures. For instance
+Add type hints to your variables, and marimo will carry over these types
+hints to cells where these variables use. This, combined with declaring
+your inputs in the setup cell, makes it possible for your editor
+to give completions on cell inputs.
+
+
+For example:
 
 ```python
+# setup cell
+import numpy as np
+
 # cell 1
-x: int
-y: str
+x: np.ndarray
 
 # cell 2
-z = f"{x} & {y}"
+np.mean(x)
 ```
 
 will be serialized as
 
 ```python
+import marimo
+
+app = marimo.App()
+
+with app.setup:
+    import numpy as np
+
 @app.cell
-def cell_1():
-    x: int
-    y: str
-    return x, y
+def _():
+    x: np.ndarray
+    return x,
 
 @app.cell
-def cell_2(x: int, y: str):
-    z = f"{x} & {y}"
+def _(x: np.ndarray):
+    np.mean(x)
+
+if __name__ == "__main__":
+    app.run()
 ```
 
-### Markdown file format
+### As markdown
 
 !!! tip "Markdown File format tutorial"
     Run `marimo tutorial markdown-format` at the command line for a full guide.
 
 marimo notebooks can also be stored as Markdown files. This is a good option
 for prose heavy text, and can be easy to navigate and edit in external editors.
+To convert a marimo notebook to markdown, use
+
+```
+marimo export md notebook.py -o notebook.md
+```
+
+at the command-line.
 
 marimo conforms to standard markdown document format, and will render most
-places like Github.
-Metadata in this file format is saved in the frontmatter, which marimo may use
-for information like [sandboxing](../package_reproducibility.md), and the
-marimo version. All other fields are kept, but ignored.
+places like Github. Metadata in this file format is saved in the frontmatter,
```

Comment:
Are you wrapping 80 col? I think I have break sentences and wrap 80 col

Reply:
I have a habit of mashing `gq` / `gw` to 80 col yes. I don't think it affects how the markdown is rendered, just pollutes the diff maybe.